### PR TITLE
Sauna API and Sauna Editor

### LINKED ISSUE #8: Configure Sauna Types Frontend
## Story
As an admin, I want to add all sauna types, edit their properties and add pictures to them.

## Questions
- Do we need images? Probably yes.

## Conditions
- [ ] ...

### LINKED ISSUE #43: Sauna Hook Frontend
We need a hook which provides the whole functionality for saunas

### LINKED ISSUE #47: Adjust frontend to changed backend sauna endpoints
When https://github.com/saunah/saunah-backend/issues/37 and https://github.com/saunah/saunah-backend/issues/34 are done, the frontend has to be updated to work with the new endpoints. 
Start by reverting the commits [4fd82fe](https://github.com/saunah/saunah-frontend/pull/45/commits/4fd82fe2949011928142fcbe7f125c1364a0c314) and [233eea8](https://github.com/saunah/saunah-frontend/pull/45/commits/233eea877ab67b8b00498892e6f6c9137caa2a65).

### PR DESCRIPTION
**Pull request by triumphvalley**
## Summary
- Adds the crud-api of saunas (Also adjust it to the latest backend)
- Adds a sauna-editor to edit and create saunas (Available under `/saunas/:id/edit` and `saunas/create`)
- Note: The editor takes up the complete size of the div, which looks stupid. We probably need a container element for the pages. We can wrap the view with this container when we have one.
- The hooks have been deleted on the branch, they are currently not needed

## Related Issues

- Closes #47 
- Closes #43 
- Closes #8

- Continuation of #45 

## Definition of Done

- User story
  - [x] All stated conditions fulfilled
- Code
  - [x] No more code needed
  - [x] No known bugs
- Clean Code
  - [x] Documentation for functions containing logic and components has been added
- Testing
  - [x] All existing tests pass
  - [x] New unit tests created according to [CONTRIBUTING.md](./CONTRIBUTING.md#-testing-strategy)
  - [x] New unit tests pass
- SonarCloud
  - [x] Quality gate passed
- Compatibility to backend
  - [x] Compatibility to staging-backend ([https://saunah-backend-staging.k8s.init-lab.ch](https://saunah-backend-staging.k8s.init-lab.ch)) verified
- Not behind `main`
  - [x] This branch is not behind `main` branch, (current `main` is merged into this branch)

### CODE REVIEW

File: src/utils/mapping.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/**
```
Comment (simonschuhmacher) [inline src/utils/mapping.ts:1]: I think we should discuss this module, as stated above 🙂

File: src/networking/apiRoutes.ts [hunk 2]
```diff
@@ -1,5 +1,13 @@
 import { readonly } from '../utils/object'
 
-const apiRoutes = {}
+const apiRoutes = {
+    sauna: {
+        list: '/saunas',
+        get: (id: number) => `/sauna/${id}`,
+        add: '/sauna',
```
Comment (simonschuhmacher) [inline src/networking/apiRoutes.ts:7]: I think this is `sauna/add` in the latest version of the backend.

File: src/views/routes/sauna/SaunaEditorView.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { useParams } from 'react-router-dom'
```
Comment (simonschuhmacher) [inline src/views/routes/sauna/SaunaEditorView.tsx:1]: Editing seems not to work for me for latest backend (manually disabled authentication for the route to test it):
```
Sauna.ts:68 Uncaught (in promise) Error: Object [object Object] could not be mapped in. It is not of type RemoteResponse.
    at Object.mapIn (Sauna.ts:68:1)
    at Object.get (sauna.ts:22:1)
```

**Review by simonschuhmacher — CHANGES_REQUESTED** [review]
Generally really nice, some things noted in the comments :)

File: src/views/routes/sauna/SaunaEditorView.tsx (hunk 3 above)
Comment (triumphvalley) [reply src/views/routes/sauna/SaunaEditorView.tsx:1]: I tested it manually, should work now

**Review by simonschuhmacher — APPROVED** [review]
Das händ Sie fein glöst! Lg G.Gr.

**Pull request merged**